.github/workflows/main.yml

```diff
@@ -49,19 +58,49 @@ jobs:
             libvorbis-dev \
             libspdlog-dev \
             libglew-dev \
-            libgtest-dev
+            libgtest-dev \
+            ccache
 
-      - name: Install dependencies with vcpkg
+      - name: Override cache folder for vcpkg
         if: ${{ matrix.os == 'windows-2019' }}
+        shell: bash
         run: |
+          mkdir vcpkg-cache
           vcpkg update
+          vcpkg install --dry-run zlib glfw3 libogg libvorbis libtheora spdlog glew openal-soft bullet3 fmt gtest tinyxml2 glm opengl >> vcpkg-cache/vcpkg_dry_run.txt
+
+      - name: Cache dependencies for vcpkg
+        if: ${{ matrix.os == 'windows-2019' }}
+        uses: actions/cache@v3
+        with:
+          path: ${{ github.workspace }}/vcpkg-cache
+          key: ${{ matrix.os }}-build-vcpkg-cache-${{ github.ref_name }}-${{ hashFiles('vcpkg-cache/*') }}
```

Comment:
Couldn't you just go over the vcpkgs root directory ports files, something like hashFiles('C:/vcpkg/ports/*/vcpkg.json'). Then the dry run in the step before wouldn't be necessary. I admit, thati do not know what would be more efficient